pkg/agent/proxy/proxier.go

```diff
@@ -976,6 +990,19 @@ func NewProxier(
 		ipFamily = corev1.IPv6Protocol
 	}
 
+	var serviceHealthServer healthcheck.ServiceHealthServer
+	if proxyAllEnabled {
+		nodePortAddressesString := make([]string, len(nodePortAddresses))
+		for i, address := range nodePortAddresses {
+			if isIPv6 {
+				nodePortAddressesString[i] = fmt.Sprintf("%s/128", address.String())
+			} else {
+				nodePortAddressesString[i] = fmt.Sprintf("%s/32", address.String())
+			}
```

Comment:
There is some difference between the `nodePortAddresses` argument of Antrea's proxy and kube-proxy: the former is already the concrete Node IPs that this Node is configured with:https://github.com/antrea-io/antrea/blob/52db699a6634529e2de202565fdce70936ccea9e/cmd/antrea-agent/agent.go#L215-L221 while the latter is the user provided CIDRs that will be used to filter concrete Node IPs.

Since service health server actually needs the former, I think we could pass the slice of node IP strings to it directly, instead of passing the slice of CIDRs and calling `utilproxy.GetNodeAddresses(nodePortAddresses, utilproxy.RealNetwork{})` to get concrete Node IPs again, which should just get the same results.

In this way, the changes to util/network.go and util/utils.go are not needed.

Reply:
There will be a small defect that unlike kube-proxy listening to "0.0.0.0"/"::0" when user provided nodeAddresses is empty, antrea proxy will always listen to concrete node IPs because the `nodePortAddresses` argument are used for other purposes which need the IPs to be concrete ones (for checking whether a packet is accessing a Node IP). But it shouldn't affect the functionality of service health check.

Reply:
Thanks for the insight!